sphinx_rtd_theme/layout.html

```diff
@@ -133,23 +135,32 @@
         <div class="wy-side-nav-search" {% if theme_style_nav_header_background %} style="background: {{theme_style_nav_header_background}}" {% endif %}>
           {%- block sidebartitle %}
 
-          {%- if logo and theme_logo_only %}
+          {%- if ( logo or logo_url ) and theme_logo_only %}
+
+          {%- if sphinx_version_info < (4, 0) -%}
+            {# the logo helper function was removed in Sphinx 6 and deprecated since Sphinx 4 #}
+            {# the master_doc variable was renamed to root_doc in Sphinx 4 (master_doc still exists in later Sphinx versions) #}
+            {%- if logo %}
             <a href="{{ pathto(master_doc) }}">
+              <img src="{{ pathto('_static/' + logo, 1) }}" class="logo" alt="{{ _('Logo') }}"/>
+            </a>
+            {%- endif %}
+          {# Behavior since Sphinx 4 #}
           {%- else %}
-            <a href="{{ pathto(master_doc) }}" class="icon icon-home"> {{ project }}
+            {% if logo_url %}
+              <a href="{{ pathto(root_doc) }}">
+                <img src="{{ logo_url }}" class="logo" alt="{{ _('Logo') }}"/>
+              </a>
+            {%- endif %}
           {%- endif %}
 
-          {%- if logo %}
-            {#- Not strictly valid HTML, but it's the only way to display/scale
-                it properly, without weird scripting or heaps of work
-            #}
+          {%- else %}
```

Comment:
It seems like `theme_logo_only` option wouldn't operate correctly with this logic. The conditional above is `if (logo or logo_url) and theme_logo_only`, but the `else` clause here doesn't output a logo.

That is, the current logic is meant to be:

```
<a href="...">
  {% if not theme_logo_only %}{{ project }}{% endif %}
  <img src="{{ logo }}" />
</a>
```

But if I'm interpreting the template logic here correctly, it's instead:

```
<a href="...">
  {% if theme_logo_only %}
    <img src="{{ logo }}" />
  {% else %}
    {{ project }}
  {% endif %}
</a>
```

Reply:
That `a` tag has `class="icon icon-home"` when displaying `{{ project }}`

Reply:
The logic that is "lost" from the **old** version is the ability to set `theme_logo_only=False` and get just the logo with `<a href="{{ pathto(master_doc) }}" class="icon icon-home"> {{ project }}` in the start. Which I think isn't desirable in any case.

Old version:

```
{%- if logo and theme_logo_only %}
  <a href="{{ pathto(master_doc) }}">
{%- else %}
  <a href="{{ pathto(master_doc) }}" class="icon icon-home"> {{ project }}
{%- endif %}

{%- if logo %}
  {#- Not strictly valid HTML, but it's the only way to display/scale
      it properly, without weird scripting or heaps of work
  #}
  {%- if sphinx_version_info < (4, 0) -%}
  <img src="{{ pathto('_static/' + logo, 1) }}" class="logo" alt="{{ _('Logo') }}"/>
  {%- else %}
  <img src="{{ logo_url }}" class="logo" alt="{{ _('Logo') }}"/>
  {%- endif %}
{%- endif %}
</a>
```

Reply:
> Which I think isn't desirable in any case.

That is actually the intended use of that option though:
https://sphinx-rtd-theme.readthedocs.io/en/stable/configuring.html#confval-logo_only

The options users have are:

- No logo, text only
- Logo and text (`logo_only = False`, the default)
- Logo only (`logo_only = True`)

Reply:
Gotcha, it requires a bit of extra imagination :innocent:  but definitely a flexible configuration option that makes it possible to have all combinations :+1:

Reply:
The new version is a bit different, in some way more condensed than the former version wrt `theme_logo_only`.

Reply:
New version looks good! It was much faster to verify too, and I've only had one cup of coffee even :coffee: